pkg/controller/secret-mirror.go

```diff
@@ -99,7 +100,23 @@ func (c *SecretMirror) Run(workers int, stopCh <-chan struct{}) {
 		go wait.Until(c.worker, time.Second, stopCh)
 	}
 
-	<-stopCh
+	select {
+	case change := <-changes:
+		start := time.Now()
+		if err := c.handleChange(change); err != nil {
+			logrus.WithError(err).Error("Error handle change.")
+		}
+		logrus.WithField("duration", fmt.Sprintf("%v", time.Since(start))).Info("change handled")
+	case <-stopCh:
+		logrus.Info("controller is shutting down...")
+	}
+}
+
+func (c *SecretMirror) handleChange(delta config.Delta) error {
```

Comment:
Yeah just dynamically load the config when you need it and you will not need to do anything here. The periodic relist will handle it.

Reply:
Done.